port/cpl_vsisimple.cpp

```diff
@@ -843,10 +858,10 @@ void *VSIRealloc(void *pData, size_t nNewSize)
 void VSIFree(void *pData)
 
 {
-#ifdef DEBUG_VSIMALLOC
-    if (pData == nullptr)
+    if (pData == nullptr || pData == VSI_STATIC_EMPTY_STRING)
```

Comment:
```suggestion
    if (pData == nullptr || pData == VSIGetStaticEmptyString())
```

Reply:
Well, checking against VSIGetStaticEmptyString() involves a function call in which a static variable must be checked for being initialized in a thread-safe way. This will definitely have some cost compared to the approach to comparing to a static variable initialized at module loading.

Reply:
Yes, there is a price tag, and it may prevent inlining.

It shouldn't be too hard to initialize a global-scope VSI_STATIC_EMPTY_STRING with a dynamic allocation without running in static initialization order fiasko.

But avoid entering undefined behaviour with a casting away constness from truly const memory.

(If it only wasn't too invasive to simply use nullptr instead of a empty strings...)

Reply:
Actually nothing dynamic needed.